httputil/cert_ext_tripper.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package httputil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type ignoreExtensionsRoundTripper struct {
+	base         *http.Transport
+	extsToIgnore []asn1.ObjectIdentifier
+	logger       hclog.Logger
+}
+
+// NewIgnoreUnhandledExtensionsRoundTripper creates a RoundTripper that may be used in an HTTP client which will
+// ignore the provided extensions if present unhandled on a certificate.  If base is nil, the default RoundTripper is used.
+func NewIgnoreUnhandledExtensionsRoundTripper(logger hclog.Logger, base http.RoundTripper, extsToIgnore []asn1.ObjectIdentifier) http.RoundTripper {
+	if len(extsToIgnore) == 0 {
```

Comment:
Shouldn't this error instead? It seems like it should be an error to be using this lib when you aren't ignoring anything.

Reply:
Yeah, I think that's probably cleaner.